tools/prow-config-updater/github_commenter.go

```diff
@@ -0,0 +1,148 @@
+			return err
+		},
+		ghc.dryrun,
+	)
+}
+
+// Comment on the pull request for the Prow update failure result.
+func (ghc *GitHubCommenter) commentOnUpdateConfigsFailure(prnumber int, env config.ProwEnv, files []string, err error) error {
+	comment := fmt.Sprintf(failureUpdatedConfigsCommentTemplate, env, fileListCommentString(files), err)
+	return helpers.Run(
+		fmt.Sprintf("Creating 'update failure' comment %q on PR %q in %s/%s",
+			comment, prnumber, config.OrgName, config.RepoName),
+		func() error {
+			_, err := ghc.client.CreateComment(config.OrgName, config.RepoName, prnumber, comment)
+			return err
+		},
+		ghc.dryrun,
+	)
+}
+
+// Try to get the fork pull request tested and merged by staging Prow.
+func (ghc *GitHubCommenter) tryMergeForkPullRequest(forkOrgName string, prnumber int) error {
+	commentToAdd := "/ok-to-test\n/lgtm\n/approve"
+	labelToAdd := []string{"cla: yes"}
+	return helpers.Run(
+		fmt.Sprintf("Add comment %q on the pull request", commentToAdd),
+		func() error {
+			errs := make([]error, 2)
+			_, errs[0] = ghc.client.CreateComment(forkOrgName, config.RepoName, prnumber, commentToAdd)
+			errs[1] = ghc.client.AddLabelsToIssue(forkOrgName, config.RepoName, prnumber, labelToAdd)
+			return helpers.CombineErrors(errs)
+		},
+		ghc.dryrun,
+	)
+}
+
+// Comment on the pull request for the staging success result.
+func (ghc *GitHubCommenter) commentOnStagingSuccess(prnumber int, err error) error {
+	return helpers.Run(
+		fmt.Sprintf("Creating 'staging success' comment %q on PR %q in %s/%s",
+			successStagingCommentTemplate, prnumber, config.OrgName, config.RepoName),
+		func() error {
+			_, err := ghc.client.CreateComment(config.OrgName, config.RepoName, prnumber, successStagingCommentTemplate)
+			return err
+		},
+		ghc.dryrun,
+	)
+}
+
+// Comment on the pull request for the staging failure result.
+func (ghc *GitHubCommenter) commentOnStagingFailure(prnumber int, err error) error {
+	comment := fmt.Sprintf(failureStagingCommentTemplate, err)
+	return helpers.Run(
+		fmt.Sprintf("Creating 'staging failure' comment %q on PR %q in %s/%s",
+			comment, prnumber, config.OrgName, config.RepoName),
+		func() error {
+			_, err := ghc.client.CreateComment(config.OrgName, config.RepoName, prnumber, comment)
+			return err
+		},
+		ghc.dryrun,
+	)
+}
+
+// Comment on the pull request for the rollout success result.
+func (ghc *GitHubCommenter) commentOnRolloutSuccess(prnumber, newprnumber int) error {
+	comment := fmt.Sprintf(successRolloutCommentTemplate, newprnumber)
+	return helpers.Run(
+		fmt.Sprintf("Creating 'rollout success' comment %q on PR %q in %s/%s",
+			comment, prnumber, config.OrgName, config.RepoName),
+		func() error {
+			_, err := ghc.client.CreateComment(config.OrgName, config.RepoName, prnumber, comment)
+			return err
+		},
+		ghc.dryrun,
+	)
+}
+
+// Comment on the pull request for the rollout failure result.
+func (ghc *GitHubCommenter) commentOnRolloutFailure(prnumber int, err error) error {
```

Comment:
Now I really feel like there should be a helper function :)

Reply:
Added helper function :)